backend/lib/endpoints/posts.js

```diff
@@ -30,21 +30,25 @@ async function routes(app) {
   app.get(
     "/",
     {
-      preValidation: [app.authenticate],
       schema: getPostsSchema,
     },
     async (req) => {
       const { userId } = req.query;
```

Comment:
what is the point of `userId` in the query param? isn't this to get the posts posted by a specific user?
I thought posts was always a public route, but seems like here this is different for some filters, specifically location.
Putting id in query is not secure if you put another user's id? But I might not understand what's happening here.
I think @joshmorel his suggestion to add `optional` param to the `authenticate` hook is actually a great suggestion, or maybe `app.optionalAuthenticate` as alternative or something similar, so we can add `userId` on the request if available

Reply:
The posts are returned sorted by distance to the user at the `/posts` endpoint.
For that, if the user is authenticated, we get the user's location to serve as a reference point (to calculate distance and etc).

And I agree that an "optional" param or another hook would be the best way here. 
This endpoint is always public, but the behavior changes based on whether the user is authenticated or not (if he is, sort by his location, otherwise it users other sorting criteria).

Reply:
And about not being safe, yeah, ideally that would be handled by the optional auth hook, but even with it as query param, it is not a big problem here since the id is only used here to decide there reference point for sorting the posts (putting another user id will just change posts order).

Reply:
ah ok ... well not sure if we want to allow anyone to paste anyone else's user id in there and see what it returns? :) maybe i'm thinking too far but you never know how someone could abuse this kind of things...

Reply:
Yeah, ideally we probably don't want that, but until we have an optional auth hook that may be the best way (and with no major side effects). 

So I suggest leaving it like this until we have the hook and then we switch.

Reply:
lgtm (for now). I think adding a new ticket for optional auth hook is a good idea. I'll do so.

Reply:
Please do so and link to this so we can remember do update it here